### PR TITLE
added mocks for external services in lambda_alerting tests

### PR DESCRIPTION
**Pull request by natmasslove**

### CODE REVIEW

**Comment by natmasslove** [comment]
This tiny PR is just to dip my toe's in a muddy water of pytesting :) Specifically, using mocks to ensure no calls to AWS are made.

So, please checks if it's movement in a right direction. Thanks!

**Comment by natmasslove** [comment]
image

Doesn't look to embarrasing :)

File: tests/src/test_lambda_alerting.py [hunk 1]
```diff
@@ -63,6 +63,43 @@ def event_dyn_props_init(aws_props_init):
     return (account_id, region, time_str, epoch_time, version_str)
 
 
+################################################################################################################################
+
+# mocking AWS calls done in lambda_alerting
+
+
+@pytest.fixture(scope="function", autouse=True)
```
Comment (npaskrobka) [inline tests/src/test_lambda_alerting.py:71]: I would think on the scope, probably you can use a broader scope like "module" or "session" to improve efficiency.
Now scope "function" means mocks will be executed for each test function.

**Review by npaskrobka — COMMENTED** [review]
Looks very good!
I would advice to test also notification messages returned by lambda (check that result['messages'] expected)

**Review by aryna-yukhnevich — APPROVED** [review]

File: tests/src/test_lambda_alerting.py [hunk 2]
```diff
@@ -39,16 +39,16 @@ def aws_props_init():
...
-    ] = f"log-group-salmon-alert-events-{stage_name}"
-    os.environ[
-        "ALERT_EVENTS_CLOUDWATCH_LOG_STREAM_NAME"
-    ] = f"log-stream-salmon-alert-events-{stage_name}"
+    os.environ["ALERT_EVENTS_CLOUDWATCH_LOG_GROUP_NAME"] = (
+        f"log-group-salmon-alert-events-{stage_name}"
+    )
+    os.environ["ALERT_EVENTS_CLOUDWATCH_LOG_STREAM_NAME"] = (
+        f"log-stream-salmon-alert-events-{stage_name}"
+    )
```
[the remark this answers is not in the document]
Comment (natmasslove) [reply tests/src/test_lambda_alerting.py]: Good catch. It's no longer needed. Replace with dummy value for stage-name

File: tests/src/test_lambda_alerting.py (hunk 1 above)
Comment (natmasslove) [reply tests/src/test_lambda_alerting.py]: Agree. Changed to module scope.
Had to switch to unittest.mock functionality, as pytest wrapper doesn't allow anything but "function" scope

**Review by npaskrobka — APPROVED** [review]

**Pull request merged**